File: DesafioAutomacao/src/main/java/com/qa/blogdoagi/pages/MainPage.java
package com.qa.blogdoagi.pages;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import static com.microsoft.playwright.assertions.PlaywrightAssertions.assertThat;

import com.microsoft.playwright.FrameLocator;
import com.microsoft.playwright.options.AriaRole;
import com.microsoft.playwright.options.LoadState;

public class MainPage {
	private Page page;
	private Locator iconePesquisa;
	private Locator campoSearch;
	private Locator elementoH1;
	private Locator menuProduto;
	private Locator menuEmprestimo;
	private Locator subMenuEmprestimo;
	private Locator campoEmail;
	private Locator botaoInscrever;
	private Locator botaoEntendi;
	private Locator botaoPaginacao;

	public MainPage(Page page) {

		this.page = page;
		this.iconePesquisa = page.getByLabel("Link do ícone de pesquisa");
		this.campoSearch = page.getByPlaceholder("Search...");
		this.elementoH1 = page.locator("h1");
		this.menuProduto = page.locator("//span[text()='Produtos']");
		this.menuEmprestimo = page.locator("//span[text()='Empréstimos']");
		this.subMenuEmprestimo = page.getByRole(AriaRole.LINK,
				new Page.GetByRoleOptions().setName("Empréstimo na Conta de Luz"));
		this.campoEmail = page.getByPlaceholder("Adicione seu e-mail ");
		this.botaoInscrever = page.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Inscrever-se"));
		this.botaoEntendi = page.frameLocator("internal:text=\"This feature requires inline\"i")
				.getByRole(AriaRole.BUTTON, new FrameLocator.GetByRoleOptions().setName("Entendi"));
		this.botaoPaginacao = page.getByRole(AriaRole.LINK, new Page.GetByRoleOptions().setName("2").setExact(true));

	}

	public void clicarBotaoPesquisa() {
		page.waitForLoadState(LoadState.LOAD);
		page.waitForLoadState();
		iconePesquisa.click();

	}

	public void inserirPalavra() {
		campoSearch.fill("produtos");
		campoSearch.press("Enter");

	}

	public void validarPalavraPesquisada() {
		page.waitForLoadState(LoadState.LOAD);
		assertThat(elementoH1).containsText("Resultados encontrados para: produtos");

	}

	public void focarMenuEmprestimo() {
		menuProduto.first().hover();
		menuEmprestimo.first().hover();
	}

	public void clicarMenuEmprestimo() {
		subMenuEmprestimo.click();
	}

	public void validarMensagemEmprestimo() {
		page.waitForLoadState(LoadState.LOAD);
		assertThat(elementoH1).containsText("Empréstimo na Conta de Luz");
	}

	public void preencherEmail() {
		page.waitForLoadState(LoadState.LOAD);
		campoEmail.fill("test@email.com");

	}

	public void clicarBotaoInscrever() {
		botaoInscrever.click();
		page.waitForTimeout(2000);

	}

	public void clicarBotaoEntendi() {
		botaoEntendi.click();
	}

	public void validarAusenciaPopUpInscricao() {
		page.waitForTimeout(2000);
		assertThat(botaoEntendi).isHidden();
	}

	public void clicarBotaoPaginacao() {
		page.waitForTimeout(3000);
		botaoPaginacao.click();

	}

	public void validarUrlPaginacao2() {
		page.waitForTimeout(3000);
		assertThat(page).hasURL("https://blogdoagi.com.br/page/2/");
	}

	public void validarBotaoAPPStore() {
		page.waitForLoadState(LoadState.LOAD);
		Page page1 = page.waitForPopup(() -> {
			page.locator(".wp-block-image > a").first().click();
		});
		page1.waitForTimeout(1000);
		assertThat(page1.locator("h1")).containsText("Agibank: Conta, Crédito Brazil L +4");
	}

}

File: Pages/MainPage.cs
using Microsoft.Playwright;
using Microsoft.Playwright.NUnit;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PlaywrightTests.Pages
{
    public class MainPage : PageTest
    {
        private IPage _page;
        private ILocator iconeDePesquisa;
        private ILocator campoSearch;
        private ILocator elementoH1;
        private ILocator menuProduto;
        private ILocator menuEmprestimo;
        private ILocator subMenuEmprestimo;
        private ILocator campoEmail;
        private ILocator botaoInscrever;
        private ILocator botaoEntendi;
        private ILocator botaoPaginacao;

        
        public MainPage(IPage page)
        {
            _page = page;
            iconeDePesquisa = _page.GetByLabel("Link do ícone de pesquisa");
            campoSearch = _page.GetByPlaceholder("Search...");
            elementoH1 = _page.Locator("h1");
            menuProduto = _page.Locator("//span[text()='Produtos']");
            menuEmprestimo = _page.Locator("//span[text()='Empréstimos']");
            subMenuEmprestimo = _page.GetByRole(AriaRole.Link, new() { Name = "Empréstimo na Conta de Luz" });
            campoEmail = _page.GetByPlaceholder("Adicione seu e-mail ");
            botaoInscrever = _page.GetByRole(AriaRole.Button, new() { Name = "Inscrever-se" });
            botaoEntendi = _page.GetByText("This feature requires inline").Locator("internal:control=enter-frame").GetByRole(AriaRole.Button, new() { Name = "Entendi" });
            botaoPaginacao = _page.GetByRole(AriaRole.Link, new() { Name = "2", Exact = true });
        }

        public async Task ClickBotaoPesquisa()
        {
            await _page.WaitForLoadStateAsync(LoadState.Load);
            await iconeDePesquisa.ClickAsync();
        }

        public async Task InserirPalavra()
        {
            await campoSearch.FillAsync("produtos");
            await campoSearch.PressAsync("Enter");
        }

        public async Task ValidarPalavraPesquisada()
        {
            await _page.WaitForLoadStateAsync(LoadState.Load);
            await Expect(elementoH1).ToContainTextAsync("produtos");
        }

        public async Task FocarMenuEmprestimo()
        {
            await menuProduto.First.HoverAsync();
            await menuEmprestimo.First.HoverAsync();
        }

        public async Task ClicarMenuEmprestimo() => await subMenuEmprestimo.ClickAsync();

        public async Task ValidarMensagemEmprestimo()
        {
            await _page.WaitForLoadStateAsync(LoadState.DOMContentLoaded);
            await Expect(elementoH1).ToContainTextAsync("Empréstimo na Conta de Luz");
        }

        public async Task PreencherEmail()
        {
            await _page.WaitForLoadStateAsync(LoadState.Load);
            Thread.Sleep(4000);
            await campoEmail.FillAsync("test@gmail.com");
        }

        public async Task ClicarBotaoInscrever() => await botaoInscrever.ClickAsync();

        public async Task ClicarBotaoEntendi()
        {
            await _page.WaitForLoadStateAsync(LoadState.DOMContentLoaded);            
            await botaoEntendi.ClickAsync();
        }

        public async Task ValidarAusenciaPopUpInscricao()
        {
            await _page.WaitForLoadStateAsync(LoadState.Load);
            Thread.Sleep(4000);
            await Expect(elementoH1).ToBeHiddenAsync();
        }

        public async Task ClicarBotaoPaginacao()
        {
            await _page.WaitForLoadStateAsync(LoadState.Load);
            await botaoPaginacao.ClickAsync();
        }

        public async Task ValidarURLPaginacao()
        {
            await _page.WaitForLoadStateAsync(LoadState.Load);
            await Expect(_page).ToHaveURLAsync("https://blogdoagi.com.br/page/2/");
        }

        public async Task ValidarBotaoAppSotre()
        {
            await _page.WaitForLoadStateAsync(LoadState.Load);
            var page1 = await _page.RunAndWaitForPopupAsync(async () =>
            {
                await _page.Locator(".wp-block-image > a").First.ClickAsync();
            });
            await Expect(page1.Locator("h1")).ToContainTextAsync("Agibank: Conta, Crédito Brazil L +4");
        }
    }
}
